### PR TITLE
tuxclocker: 1.4.0 -> 1.5.0

### PR DESCRIPTION
**Pull request by Lurkki14**
## Description of changes
- git is no longer required to build plugins
- Way to disable building CPU plugin was added upstream, so patch is no longer needed
- Added `passthru.tests` so Ofborg builds `tuxclocker-without-unfree`

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/tu/tuxclocker-plugins/package.nix
Comment (dotlambda) [inline pkgs/by-name/tu/tuxclocker-plugins/package.nix:1]: Remove `git` from arguments.

File: pkgs/applications/misc/tuxclocker/default.nix [hunk 1]
```diff
@@ -56,6 +57,8 @@ stdenv.mkDerivation (finalAttrs: {
     "-Dplugins=false"
   ];
 
+  passthru.tests = pkgs.tuxclocker-without-unfree;
```
Comment (dotlambda) [inline pkgs/applications/misc/tuxclocker/default.nix:60]:
```suggestion
  passthru.tests = tuxclocker-without-unfree;
```

File: pkgs/applications/misc/tuxclocker/default.nix [hunk 2]
```diff
@@ -16,14 +17,14 @@
...
 stdenv.mkDerivation (finalAttrs: {
   pname = "tuxclocker";
-  version = "1.4.0";
+  version = "1.5.0";
 
   src = fetchFromGitHub {
     owner = "Lurkki14";
     repo = "tuxclocker";
     fetchSubmodules = true;
     rev = "${finalAttrs.version}";
```
Comment (dotlambda) [inline pkgs/applications/misc/tuxclocker/default.nix:26]:
```suggestion
    rev = finalAttrs.version;
```

File: pkgs/applications/misc/tuxclocker/default.nix [hunk 3]
```diff
@@ -7,6 +7,7 @@
 , meson
 , ninja
 , pkg-config
+, pkgs
```
Comment (dotlambda) [inline pkgs/applications/misc/tuxclocker/default.nix:10]:
```suggestion
, tuxclocker-without-unfree
```

File: pkgs/applications/misc/tuxclocker/default.nix
Comment (dotlambda) [inline pkgs/applications/misc/tuxclocker/default.nix:1]: The `git` in here is still needed though?

Comment (Lurkki14) [reply pkgs/applications/misc/tuxclocker/default.nix:1]: It's used to get version string, which isn't needed for the plugins

File: pkgs/applications/misc/tuxclocker/default.nix [hunk 4]
```diff
@@ -56,6 +57,8 @@ stdenv.mkDerivation (finalAttrs: {
     "-Dplugins=false"
   ];
 
+  passthru.tests = tuxclocker-without-unfree;
```
Comment (dotlambda) [inline pkgs/applications/misc/tuxclocker/default.nix:60]:
```suggestion
  passthru.tests = {
    inherit tuxclocker-without-unfree;
  };
```

File: pkgs/by-name/tu/tuxclocker-nvidia-plugin/package.nix [hunk 5]
```diff
@@ -22,13 +22,10 @@ stdenv.mkDerivation {
...
 
-  # Build doesn't have a way to disable building the CPU plugin, which is already
-  # provided by 'tuxclocker-plugins'
-  patches = [ ./no-cpu-plugin.patch ];
-
   mesonFlags = [
     "-Ddaemon=false"
     "-Dgui=false"
     "-Drequire-nvidia=true"
+    "-Dplugins-cpu=false"
```
Comment (dotlambda) [inline pkgs/by-name/tu/tuxclocker-nvidia-plugin/package.nix:29]:
```suggestion
    "-Dplugins-cpu=false" # provided by tuxclocker-plugins
```

**Pull request merged**